pkg/cli/project/project.go

```diff
@@ -81,6 +83,15 @@ func NewCmdProject(fullName string, f kcmdutil.Factory, streams genericclioption
 		Example: fmt.Sprintf(projectExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
 			o.PathOptions = cliconfig.NewPathOptions(cmd)
+			err := func() error {
```

Comment:
This should move to `Complete`, that's the purpose of the `Complete` method to ensure all the `ProjectOptions` struct fields are ready for the `Run` which is responsible for actual execution of the command.

Reply:
That was my original intention. Though, it will not fly without changing `Complete(f genericclioptions.RESTClientGetter, args []string) error` signature to `Complete(f genericclioptions.RESTClientGetter, args []string, cmd *cobra.Command) error` so I can read `cmd.Flags().GetString("context")` from within `Complete`. Is that acceptable?

Reply:
Yup, that's perfectly reasonable. We do that elsewhere already, only change the order, which is usually: factory, cmd, args :wink:

Reply:
Addressed, PTAL